pkg/diagnostics/analyze.go

```diff
@@ -338,6 +348,16 @@ func setNamespacetoreDetailsInJob(namespaceStore *nbv1.NamespaceStore, cmd *cobr
 			namespaceStore.Kind, namespaceStore.Name, namespaceStore.Namespace)
 	}
 
+	// in case it is aws endpoint, we would pass the region
```

Comment:
@shirady I see repeated code, can we move this into a method and refer to that method?

Reply:
It's 2 different types One part is for namespacestore and another part is for backingstore as we have two files with similar functions. The mutual code that was not related to types was moved to different functions.

Reply:
Using the interface you can pass any type and check for type and convert to the underlying type, but it might make the changes complicated,